app/assets/javascripts/uploadcare/widget/tabs/preview-tab.coffee

```diff
@@ -77,6 +86,24 @@ uploadcare.namespace 'widget.tabs', (ns) ->
             @__setState('image', {file: false})
             @element('image').attr('src', src)
             @initImage(size)
+            df.resolve()
+      .fail () =>
+        return df.reject('fail')
+
+      df.promise()
+
+    __tryToLoadVideoPreview: (file, blob) =>
+      if (
+        file.state() != 'pending' or
+          not blob.size
+      )
+        return
+
+      src = URL.createObjectURL(blob)
+      @dialogApi.always ->
+        URL.revokeObjectURL(src)
+      @__setState('video', {file: false})
```

Comment:
The state should be video only when the file can be rendered as a video. For images this is handled in `drawFileToCanvas()`.